File: Python/0003_Longest_Substring_Without_Repeating_Characters.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        usedChar = {}
        start = maxLength = 0
        for i in range(len(s)):
            if s[i] in usedChar and start <= usedChar[s[i]]:
                start = usedChar[s[i]] + 1
            else:
                maxLength = max(maxLength, i - start + 1)
            usedChar[s[i]] = i
        return maxLength
            

File: Javascript/0003_Longest_Substring_Without_Repeating_Characters.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function(s) {
    let usedChars = {}
    let start = 0
    let maxLength = 0
    for(let i=0; i<s.length; i++) {
        if(s[i] in usedChars && start <= usedChars[s[i]])
            start = usedChars[s[i]] + 1
        else 
            maxLength = Math.max(maxLength, i - start + 1)
        usedChars[s[i]] = i
    }
    return maxLength
};